types/index.d.ts

```diff
@@ -430,7 +430,7 @@ declare namespace Realm {
         function getAllSyncSessions(user: Realm.User): [Realm.App.Sync.Session];
         function getSyncSession(user: Realm.User, partitionValue: string|number|ObjectId|null) : Realm.App.Sync.Session;
         function setLogLevel(app: App, logLevel: LogLevel): void;
-        function setLogger(app: App, callback: (level: NumericLogLevel, message: string) => void): void;
+        function setLogger(app: App, callback: (level: string, message: string) => void): void;
```

Comment:
Did you notice that the `LogLevel` type is already defined? I suggest using that instead of `string`.
```suggestion
        function setLogger(app: App, callback: (level: LogLevel, message: string) => void): void;
```